src/components/hoc/withRestrictedAccess.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react/display-name */
+import React from 'react';
+import { Redirect } from 'react-router';
+import { isLoggedIn } from '../../utils/handlers/authenticationHandlers';
+
+const withRestrictedAccess = (BaseComponent, onlyForAdmin = false) => props => {
+  if (!isLoggedIn()) {
+    return <Redirect to="/login" />;
+  }
+
+ /*
+      TODO: Until admin validation is ready, all the users will be treated as admins
```

Comment:
We need a follow up card/issue about that before mergin it like that into master